src/test/java/io/codurance/pouch/ResourceControllerShould.java

```diff
@@ -1,30 +1,56 @@
 package io.codurance.pouch;
 
 import org.hamcrest.Matchers;
+import org.junit.jupiter.api.BeforeEach;
 import org.junit.jupiter.api.Test;
+import org.springframework.http.HttpStatus;
 
-import java.util.List;
+import java.util.Optional;
 
 import static java.util.Arrays.asList;
+import static org.apache.http.HttpStatus.*;
 import static org.junit.Assert.assertThat;
 import static org.mockito.Mockito.mock;
 import static org.mockito.Mockito.when;
 
 class ResourceControllerShould {
 
+    private ResourceRepository resourceRepository;
+    private ResourceController resourceController;
+
+    @BeforeEach
+    void setUp() {
+        resourceRepository = mock(ResourceRepository.class);
+        resourceController = new ResourceController(resourceRepository);
+    }
+
     @Test
     void return_list_of_all_resources() {
-        ResourceRepository resourceRepository = mock(ResourceRepository.class);
-
-        List<Resource> resourceList = asList(
+        var resourceList = asList(
                 new Resource(1, "2018-12-05 16:01:00", "Spring Data JDBC", "https://spring.io/projects/spring-data-jdbc"),
                 new Resource(2, "2018-12-05 16:02:00", "SQL Fiddle", "http://sqlfiddle.com/"),
                 new Resource(3, "2018-12-05 16:03:00", "PostgreSQL: The world''s most advanced open source database", "https://www.postgresql.org/"));
 
         when(resourceRepository.findAll()).thenReturn(resourceList);
 
-        ResourceController resourceController = new ResourceController(resourceRepository);
-
         assertThat(resourceController.findAll(), Matchers.is(resourceList));
     }
+
+    @Test
+    void return_one_specified_resource() {
+        var resource = new Resource(1, "2018-12-05 16:01:00", "Spring Data JDBC", "https://spring.io/projects/spring-data-jdbc");
+
+        when(resourceRepository.findById(1)).thenReturn(Optional.of(resource));
+
+        assertThat(resourceController.findById(1).getStatusCode(), Matchers.is(HttpStatus.valueOf(SC_OK)));
```

Comment:
Extract `resourceController.findById(1)` into a variable so that the call is only made once. Also static imports on matchers.

Reply:
✅